### PR TITLE
Homework5: Anton Kochnevski

### PR DESCRIPTION
**Pull request by Antonkoch999**
### PR checklist
Please make sure you've passed PR checklist
- [ ] my PR passes CI checks (flake8 and pycodestyle)
- [ ] my PR named according to guidelines.
- [ ] my PR contains complete tasks descriptions for all tasks
- [ ] my PR made from the fork feature branch to maxim homework# branch.
- [ ] my PR contains only code that solves appropriate homework tasks. Redundant 
files (configs, etc. ) are put to .gitignore

### CODE REVIEW

File: src/Homework5/runner.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+runner(‘func’, ‘func1’...) - вызывает все
+переданные функции
+"""
+
+
+from src.func import funcs
+
+
+def runner(*args):
+    lst_method = [el for el in dir(funcs) if '__' not in el]
```
Comment (belovmd) [inline src/Homework5/runner.py]: `a__b` is not a valid name?

File: src/Homework5/get_ranges.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    range_ = False
+    rating_list = ''
+
+    for el in lst:
+        index_ = lst.index(el)
+
+        if not range_:
+            if el == lst[-1] or lst[index_ + 1] - el != 1:
+                new_lst.append(str(el))
+                continue
```
Comment (belovmd) [inline src/Homework5/get_ranges.py]: why do we need it here?

File: src/Homework5/get_ranges.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+                range_ = True
+                continue
+        else:
+            if el == lst[-1] or lst[index_ + 1] - el != 1:
+                rating_list += str(el)
+                new_lst.append(rating_list)
+                rating_list = ''
+                range_ = False
+                continue
+            else:
```
Comment (belovmd) [inline src/Homework5/get_ranges.py]: redundant code

File: src/Homework5/ratings.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        print('File not found')
+
+
+def create_years_rating_name_file():
+
+    with open('top250.txt', 'r') as top:
+        file_years = open('years.txt', 'w')
+        years = re.findall(r'\D\d{4}\D', top.read())
+        for el in years:
+            gistogram = f'>>>{"*" * (int(el[1:-1]) - 1895)}>>>{el[1:-1]}\n'
```
Comment (belovmd) [inline src/Homework5/ratings.py]: `>>>{"*" * (int(el[1:-1]) - 1895)}>>>{el[1:-1]`   extremely complex

File: src/Homework5/binary_operations_1.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
+"""
```
Comment (belovmd) [inline src/Homework5/binary_operations_1.py:1]: tasks for binary operations

File: src/Homework5/binary_operations_2.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
+"""
```
Comment (belovmd) [inline src/Homework5/binary_operations_2.py:1]: task for binary operations

**Review by belovmd — COMMENTED** [review]
Thank you for your work. 
Please address next paragraphs from review notes [Review Notes](https://github.com/belovmd/it-academy-summer/blob/master/Homework%20review/Homework5.md):
- 7
- 12
- 15

File: src/Homework5/runner.py (hunk 1 above)
Comment (Antonkoch999) [reply src/Homework5/runner.py]: done

File: src/Homework5/get_ranges.py (hunk 2 above)
Comment (Antonkoch999) [reply src/Homework5/get_ranges.py]: done

File: src/Homework5/get_ranges.py (hunk 3 above)
Comment (Antonkoch999) [reply src/Homework5/get_ranges.py]: done

File: src/Homework5/ratings.py (hunk 4 above)
Comment (Antonkoch999) [reply src/Homework5/ratings.py]: done

File: src/Homework5/binary_operations_1.py (hunk 5 above)
Comment (Antonkoch999) [reply src/Homework5/binary_operations_1.py:1]: done

File: src/Homework5/binary_operations_2.py (hunk 6 above)
Comment (Antonkoch999) [reply src/Homework5/binary_operations_2.py:1]: done

File: src/Homework5/binary_operations_1.py [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+
+def binary(int_):
+    counter = 0
+
+    while (int_ >> counter) > 1:
+        counter += 1
+
+    if (int_ - (2 ** counter)) < (2 ** (counter + 1) - int_):
```
Comment (belovmd) [inline src/Homework5/binary_operations_1.py:14]: binary shift

File: src/Homework5/binary_operations_2.py [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
+
+def binary(int_):
+    counter = 0
+
+    while (int_ >> counter >= 1) and (int_ % 2 ** counter == 0):
+        counter += 1
+
+    return 2 ** (counter - 1)
```
Comment (belovmd) [inline src/Homework5/binary_operations_2.py:13]: binary shift

File: src/Homework5/rating.txt [hunk 9]
```diff
@@ -0,0 +1,251 @@
+�����������: ������: * - ���������� ������� � ����� �������
```
Comment (belovmd) [inline src/Homework5/rating.txt:1]: what is it?

File: src/Homework5/decorator.py [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+"""
+
+
+def dec(fun):
+
+    def wrapper(args):
+        print(f'Найти {args} число фибоначи')
+        result = fun(args)
+        with open('results.txt', 'a') as file:
+            file.write(str(result) + '\n')
```
Comment (belovmd) [inline src/Homework5/decorator.py:14]: func name and datetime can be added

File: src/Homework5/decorator.py [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+        return f'Результат вычисления: {fun(args)}'
```
Comment (belovmd) [inline src/Homework5/decorator.py:15]: very strange return statement.

File: src/Homework5/runner.py [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+runner(‘func’, ‘func1’...) - вызывает все
+переданные функции
+"""
+
+
+from src.func import funcs
+
+
+def runner(*args):
+    lst_method = [el for el in dir(funcs) if callable(getattr(funcs, el))]
```
Comment (belovmd) [inline src/Homework5/runner.py:16]: we should also check __ functions.

File: src/Homework5/top250.txt [hunk 13]
```diff
@@ -0,0 +1,251 @@
+      0000000125  1888533   9.2  The Shawshank Redemption (1994)
```
Comment (belovmd) [inline src/Homework5/top250.txt:1]: what is 125 and 1888533 - it contradicts task description

File: src/Homework5/years.txt [hunk 14]
```diff
@@ -0,0 +1,88 @@
+�����������: ���: * - ���������� ������� � ����� �����
```
Comment (belovmd) [inline src/Homework5/years.txt:1]: what is it?

**Review by belovmd — COMMENTED** [review]
7,5

**Pull request closed without merging**